### PR TITLE
setup IF condition for PSP

### PR DESCRIPTION
**Pull request by chiragkumarp**
setup IF condition for PSP in value.yaml

### CODE REVIEW

**Comment by chiragkumarp** [comment]
This is regarding original PR #905 correction. Once this PR approve then we can delete PR #905 as well. 

@cpate3 The PSP resources are still not conditionalized based on the mode setting in values.yaml.Usually this is done in Helm by surrounding the YAML content with an {{ if ...}}.

**Comment by chiragkumarp** [comment]
Can this PR get approval?

File: deployments/k8s/helm/signalfx-agent/values.yaml [hunk 1]
```diff
@@ -23,6 +23,14 @@ rollingUpdateMaxUnavailable: 1
 # Namespace to deploy agent in (Optional: Will default to release namespace)
 namespace:
 
+# Specify PSP (Pod Security Policy) (Optional: default to not using PSP mode if no values are provided by the user)
+pspMode:
+
+if $pspMode == 'restricted'
```
Comment (keitwb) [inline deployments/k8s/helm/signalfx-agent/values.yaml]: [Here](https://github.com/helm/charts/blob/master/stable/apm-server/templates/clusterrole.yaml#L1) is an example of conditionalizing the template.  This is yaml, so if statements aren't allowed here.

Comment (chiragkumarp) [reply deployments/k8s/helm/signalfx-agent/values.yaml]: Thanks Ben for giving hint. Sorry for late update. I also come across this Helm Flow control standard (https://github.com/helm/helm/blob/master/docs/chart_template_guide/control_structures.md) and updated value YAML code accordingly with IF condition. Is this sounds good to approve?

**Comment by chiragkumarp** [comment]
Thanks Ben for giving hint. Sorry for late update. I also come across this Helm Flow control standard (https://github.com/helm/helm/blob/master/docs/chart_template_guide/control_structures.md) and updated value YAML code accordingly with IF condition. Is this sounds good to approve?

**Comment by chiragkumarp** [comment]
Hello Jay, 

Can you approve this PR?

**Comment by keitwb** [comment]
@chiragkumarp This PR by itself doesn't really do anything -- can you combine it with #905 and present it as a single PR.  Also see the comments on that other PR.

**Comment by chiragkumarp** [comment]
Closing this PR as we are combining it with #905

**Pull request closed without merging**